### PR TITLE
Fix NEX-2150 no updater needed

### PR DESCRIPTION
**Pull request by SergiiTao**
Since updater scripts are deprecated we shouldn't create new ones
But currently, they are still mandatory to have.

This fix says: if an updater script is not defined let's just set the version mentioned in the manifest.

### CODE REVIEW

File: common/ext/class.Extension.php [hunk 1]
```diff
@@ -497,12 +497,13 @@ public function getConfigHeader($key)
     /**
      * @throws common_ext_ManifestException
      * @throws common_ext_ManifestNotFoundException
+     * @throws common_ext_UpdaterNotFoundException
```
Comment (wazelin) [inline common/ext/class.Extension.php:500]: I don't see a definition of the exception.
Have you forgotten to commit/push it?

**Review by wazelin — CHANGES_REQUESTED** [review]

File: common/ext/class.Extension.php (hunk 1 above)
Comment (SergiiTao) [reply common/ext/class.Extension.php:500]: indeed

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices (I cannot really say it does, as now the exceptional case is a default behavior for all the new extensions)
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Comment by SergiiTao** [comment]
> * New code is respecting best practices (I cannot really say it does, as now the exceptional case is a default behavior for all the new extensions)

This is not just for new extensions. In the extensions which are up to date already on all environments, we can delete deprecated scripts. Or we can delete it everywhere with a notice that an upgrade from old versions should be done in two stages.

File: common/ext/class.Extension.php [hunk 2]
```diff
@@ -497,12 +497,13 @@ public function getConfigHeader($key)
...
      */
     public function getUpdater(): common_ext_ExtensionUpdater
     {
         $updaterClass = $this->getManifest()->getUpdateHandler();
         if ($updaterClass === null) {
-            throw new \common_ext_ManifestException('No Updater found for ' . $this->getName());
+            throw new \common_ext_UpdaterNotFoundException('No Updater found for ' . $this->getName());
```
Comment (siwane) [inline common/ext/class.Extension.php:506]: Could u simplfy FQN please, and optionally use an alias?

File: common/ext/class.UpdateExtensions.php [hunk 3]
```diff
@@ -110,10 +110,16 @@ protected function updateExtension(common_ext_Extension $ext)
                     $currentVersion = $returnedVersion;
                 }
 
-                if ($currentVersion == $codeVersion) {
-                    $versionReport = new Report(Report::TYPE_SUCCESS, 'Successfully updated ' . $ext->getName() . ' to ' . $currentVersion);
+                if ($currentVersion === $codeVersion) {
+                    $versionReport = new Report(
+                        Report::TYPE_SUCCESS,
+                        'Successfully updated ' . $ext->getName() . ' to ' . $currentVersion
```
Comment (siwane) [inline common/ext/class.UpdateExtensions.php:116]: sprintf pls

File: common/ext/class.UpdateExtensions.php [hunk 4]
```diff
@@ -110,10 +110,16 @@ protected function updateExtension(common_ext_Extension $ext)
...
+                    );
                 } else {
-                    $versionReport = new Report(Report::TYPE_WARNING, 'Update of ' . $ext->getName() . ' exited with version ' . $currentVersion);
+                    $versionReport = new Report(
+                        Report::TYPE_WARNING,
+                        'Update of ' . $ext->getName() . ' exited with version ' . $currentVersion
```
Comment (siwane) [inline common/ext/class.UpdateExtensions.php:121]: sprintf pls

**Review by siwane — APPROVED** [review]
LGTM
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**